src/wampy.js

```diff
@@ -1180,22 +1179,17 @@ class Wampy {
                 return this._log(decodedPayload.err.message);
             }
 
-            argsList = decodedPayload.args;
-            argsDict = decodedPayload.kwargs;
-
-        } else {
-            argsList = data[4];
-            argsDict = data[5];
+            args = decodedPayload.args;
+            kwargs = decodedPayload.kwargs;
         }
 
-        let i = subscription.callbacks.length;
-        while (i--) {
-            await subscription.callbacks[i]({
-                details : options,
-                argsList: argsList,
-                argsDict: argsDict
-            });
+        const callbacksInReverseOrder = [...(subscription.callbacks || [])].reverse();
+        const callbackOptions = { details, argsList: args, argsDict: kwargs };
+
+        for (const callback of callbacksInReverseOrder) {
+            await callback(callbackOptions);
```

Comment:
And i think we can improve this a bit by using parallel processing instead of sequential. These are callbacks and we do not care what's going on there. So I think it is ok to use `await Promise.all` instead of cycle at all.